earthfile2llb/converter.go

```diff
@@ -2236,6 +2237,18 @@ func (c *Converter) internalRun(ctx context.Context, opts ConvertRunOpts) (pllb.
 			extraEnvVars = append(extraEnvVars, fmt.Sprintf("%s=\"$(cat %s)\"", envVar, secretPath))
 		}
 	}
+	// AWS credential import.
+	if opts.WithAWSCredentials {
+		if !c.ftrs.RunWithAWS {
+			return pllb.State{}, errors.New("RUN --aws requires the --run-with-aws feature flag")
```

Comment:
These checks tend to be done in the interrupter, which would then allow you to use `i.errorf` which would include an error message containing the Earthfile source location.